### ISSUE TITLE
Authorization issue with both [entryPoints.https.auth.basic] and Kubernetes ingress auth-type: "basic"

### ISSUE THREAD

**Issue by PierreBeucher** [containous/traefik#5515]
### Do you want to request a *feature* or report a *bug*?

Bug

### Did you try using a 1.7.x configuration for the version 2.0?

- [ ] Yes
- [x] No

### What did you do?

I deployed Traefik on Kubernetes and configured as follow:

In `traefik.toml`:
```
    [entryPoints.https.auth.basic]
        users = ["someuser:$xxx",]
```

Then created a **Secret** and **Ingress** such as:

```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
 name: myingress
 annotations:
   kubernetes.io/ingress.class: traefik
   ingress.kubernetes.io/auth-type: "basic"
   ingress.kubernetes.io/auth-secret: "mysecret" # contain auth for anotheruser:password 
spec:
 ...
```

### What did you expect to see?

One of the authentication to take precedence (such as Ingress auth config to take precedence other Traefik endpoint auth config)

### What did you see instead?

None of the authentication config worked, I kept receiving `401 Unauthorized` responses whatever login/password pair I was using. No error shown in Traefik pod. 

When using one or the other (but not both), authentication works as expected.

Documentation does not seem to specify behavior when setting both king of auth config.

### What version of Traefik are you using?

```
traefik:1.7.12-alpine
```

Deployed on Kubernetes using Helm charts `1.70.2`

**Comment by dduportal** [comment]
Hi @PierreBeucher , this behavior is not a bug.
What you want to achieve is not possible because the basic auth is based on headers that cannot be sent 2 times, so it ends on a loop: the basic auth are not merged, but are applied as a logical "AND", which is not possible, while you want a "OR".

We could see this as a feature request, but as v1.7 only accepts bug fixes and security fixes due to 2.0 release, you might want to migrate on 2.0 where authentication moved at the middleware level.

Alternatively, you could use your own authentication system with forward auth.

**Comment by dduportal** [comment]
We are closing the issue as a "declined feature request for 1.7". Feel free to reach our Community Forum to get help if you are stuck on 1.7, or re-open the issue with case description if you feel it's not working with 2.0. Thanks fro your interest!
